aws/service.go

```diff
@@ -1791,6 +1794,28 @@ func (r ConnectionErrRetryer) ShouldRetry(req *request.Request) bool {
 			return true
 		}
 	}
+	var awsErr awserr.Error
+	if errors.As(req.Error, &awsErr) {
+
+		// 	> select * from aws_region
+		// Error: NoCredentialProviders: no valid providers in chain. Deprecated.
+		// For verbose messaging see aws.Config.CredentialsChainVerboseErrors
+
+		// AWS GO SDK throws this error in case it could not find any valid credentials from all the possible methods
+		// 1. Steampipe aws config
+		// 2. AWS Environment variables
+		// 3. AWS assume role credentials
+		// 4. AWS profile
+		// 5. AWS SSO credentials
+
+		// awsErr.OrigErr()="Put "http://169.254.169.254/latest/api/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
+		// awsErr.OrigErr()="Get "http://169.254.169.254/latest/meta-data/iam/security-credentials/": dial tcp 169.254.169.254:80: connect: no route to host"
+
+		// If the error is because of invalid credentails - we should not retry
+		if strings.Contains(awsErr.OrigErr().Error(), "http://169.254.169.254/latest") {
```

Comment:
This is much broader than the comments above. Are we catching too much here? What other cases are there?

e.g. I know that timeouts do happen with the metadata service - we see this fairly frequently in Turbot. Are we going to fail too early in many cases by never retrying those?

This code / sequence is used all the time, so keen to make sure it is robust.